autosklearn/data/validation.py

```diff
@@ -0,0 +1,297 @@
+        for i, column in enumerate(X.columns):
+            if X[column].dtype.name in self.valid_pd_enc_dtypes:
+                enc_columns.append(i)
+                feature_types.append('categorical')
+            elif not np.issubdtype(X[column].dtype, np.number):
+                if X[column].dtype.name == 'Object':
+                    raise ValueError(
+                        "Input Column {} has invalid type object. "
+                        "Cast it to a numerical value before using it in Auto-Sklearn."
+                        "You can use pandas.to_numeric() to do this".format(
+                            column,
+                        )
+                    )
+                elif pd.core.dtypes.common.is_datetime_or_timedelta_dtype(
+                    X[column].dtype
+                ):
+                    raise ValueError(
+                        "Input Column {} has invalid time-type. "
+                        "Please convert the time information to a categorical value. "
+                        "Usually, time information can be separated into a set of categorical "
+                        "features. For example, hour of the day can be converted to 24 "
+                        " categories.".format(
+                            column,
+                        )
+                    )
+                else:
+                    raise ValueError(
+                        "Input Column {} has unsupported dtype {}. "
+                        "Supported column types are categorical/bool/numerical dtypes. "
+                        "Make sure your data is formatted in a correct way, "
+                        "before feeding it to Auto-Sklearn.".format(
+                            column,
+                            X[column].dtype.name,
+                        )
+                    )
+            else:
+                feature_types.append('numerical')
+
+        # Make sure we only set this once. It should not change
+        if not self.feature_types:
+            self.feature_types = feature_types
+
+        # This proc has to handle multiple calls, for X_train
+        # and X_test scenarios. We have to make sure also that
+        # data is consistent within calls
+        if enc_columns:
+            if self.enc_columns and self.enc_columns != enc_columns:
+                raise ValueError(
+                    "Validation was already performed on input with enc columns={} "
+                    "yet, another set of features with columns to encode={} was "
+                    "provided. Feature changing after fit is not allowed.".format(
+                        enc_columns,
+                        self.enc_columns
+                    )
+                )
+            else:
+                self.enc_columns = enc_columns
+
+            if not self.feature_encoder:
+                if self.feature_encoder_type == 'OneHotEncoder':
+                    encoder = preprocessing.OneHotEncoder(sparse=False)
```

Comment:
Could you please give a reason why to use the one hot encoder here?

Reply:
As per discussed offline, I will use the label encoder for the target and the ordinal encoder for the features.
No need to give flexibility here, and when needed, the user can pre-encode the values by himself/herself.